game/src/Map.cpp

```diff
@@ -0,0 +1,79 @@
+#include <include/Map.hpp>
+
+Map::Map(const uint16_t &id, const uint8_t &currentType, const std::string &mapName, const std::string &bitmapName, const tileArray &tiles, const uint16_t &dimensionX, const uint16_t &dimensionY, const entityMap &entities) {
+    setId(id);
+    setCurrentType(currentType);
+    setMapName(mapName);
+    setBitmapName(bitmapName);
+    setTiles(tiles, dimensionX, dimensionY);
+    setEntities(entities);
+
+}
+
+Map::~Map() {
+    entities_.clear();
+    
+}
+
+void Map::setId(const uint16_t &id) {
+    this->id_ = id;
+}
+void Map::setCurrentType(const uint8_t &currentType) {
+    this->currentType_ = currentType;
+}
+void Map::setMapName(const std::string &mapName) {
+    this->mapName_ = mapName;
+}
+void Map::setBitmapName(const std::string &bitmapName) {
+    this->bitmapName_ = bitmapName;
+}
+void Map::setTiles(const tileArray &tiles, const uint16_t &dimensionX, const uint16_t &dimensionY) {
+    if(dimensionX == 0 || dimensionY == 0) {
```

Comment:
Shouldn't this be an error? Can you think of a case where it's okay to have any dimension size be 0?